sadu-queries2/src/main/java/de/chojo/sadu/queries/configuration/ConnectedQueryConfiguration.java

```diff
@@ -0,0 +1,63 @@
+/*
+ *     SPDX-License-Identifier: LGPL-3.0-or-later
+ *
+ *     Copyright (C) RainbowDashLabs and Contributor
+ */
+
+package de.chojo.sadu.queries.configuration;
+
+import de.chojo.sadu.mapper.RowMapperRegistry;
+import de.chojo.sadu.queries.exception.WrappedQueryExecutionException;
+import de.chojo.sadu.queries.stages.QueryImpl;
+
+import javax.sql.DataSource;
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.util.function.Consumer;
+
+public class ConnectedQueryConfiguration extends QueryConfiguration implements AutoCloseable {
+    private Connection connection = null;
+
+    ConnectedQueryConfiguration(QueryImpl query, DataSource dataSource, boolean atomic, boolean throwExceptions, Consumer<SQLException> exceptionHandler, RowMapperRegistry rowMapperRegistry) {
+        super(query, dataSource, atomic, throwExceptions, exceptionHandler, rowMapperRegistry);
+    }
+
+    public ConnectedQueryConfiguration forQuery(QueryImpl query) {
+        return new ConnectedQueryConfiguration(query, dataSource, atomic, throwExceptions, exceptionHandler, rowMapperRegistry);
+    }
+
+    @Override
+    public void close() {
+        try {
+            if (connection != null && !connection.isClosed()) {
+                if (query.exceptions().isEmpty()) {
+                    if (atomic()) {
+                        connection.commit();
+                    }
+                }
+                connection.close();
+            }
+        } catch (SQLException e) {
+            handleException(e);
+        }
+    }
+
+    @Override
+    public Connection connection() {
+        if (connection == null) {
+            try {
+                connection = dataSource.getConnection();
+                connection.setAutoCommit(!atomic);
+            } catch (SQLException e) {
+                handleException(e);
+                throw (WrappedQueryExecutionException) new WrappedQueryExecutionException(e.getMessage()).initCause(e);
+            }
+        }
+        return connection;
+    }
+
+    @Override
+    public boolean hasConnection() {
+        return true;
```

Comment:
Its that true everytime ?

Reply:
Yes, because the connection is either created, will be created or the call to retrieve it throws an exception. This is kind of like a type check actually.

Reply:
From my point of view, this can confuse beginners with the Lib. That is why I have added this point